### ISSUE TITLE
[CLI]: Wandb syncs different local runs to the same web app run

### ISSUE THREAD

**Issue by pierclgr** [wandb/wandb#3682]
### Describe the bug

I have a python code that, by using CPU only, creates an augmented training dataset. After this code is executed and completed, another python code starts a regular training on GPUs using the augmented dataset. wandb.init() is called in this last code, not in the augmentation one.

I also have a bash script that runs the augmentation script followed by the training script and I'm running a sweep on this bash script to tune the hyperparameters of the dataset augmentation (I have parameters like brightness, contrast, blur etc. that I want to tune). 

Wandb web app perfectly detects the run since the augmentation phase, even if I'm not using wandb.init in that phase, but it happens that those runs are considered as "crashed" by the wandb web app after a while even if they're still going on and they didn't crash locally.

Then, when the training starts, the run is recovered by wandb web app and it appears as running in the web app, however wandb syncs those local runs to previous web app runs, overwriting them. 

For example, if I have 36 possible combinations of hyper parameters, what I'll have at the end of the sweep is 36 local runs (with their 36 local folders) but I'll have less than half web app runs, because wandb syncs multiple local runs to the same remote (web app) runs. I can confirm the overwriting because by looking at the logs of these runs, I'll see that all of them have been synced to the same remote run, thus all will have the same remote id and the same wandb web app link.

The number of runs that is synced together is kinda random, in most of the cases is every two runs but sometimes is also happening every three or four.

What can I do to fix this? I cannot move wandb.init from the training script to the augmentation script for many reasons, moreover this didn't happen until 5/6 days ago, runs were still reported as crashed (very rarely though) and they were not overwritten but only considered as crashed.

The command that I use for the sweep is the following (we use slurm to launch jobs on our servers):
sbatch -c 12 --mem=16G --gpus=0 --time=7-00:00:00 --array=1-50%4 wandb agent --count 1 project/sweep_id

For all these runs, there's no error in the logs and the runs finish successfully locally, it's just a weird sync problem.

### Environment

WandB version: 0.12.16

OS: Ubuntu 20.04.4 LTS

Python version: 3.9.12

**Comment by vanpelt** [comment]
Hey @pierclgr how are you calling `wandb.init`?  The only way I can think of runs being overwritten in the web app is if you're specifying an `id`.  The id must be unique for every run.  If any jobs shared the same `id` they would overwrite eachother.

**Comment by vanpelt** [comment]
The id is also set by the sweep agent in the WANDB_RUN_ID environment variable.  If your single agent is actually doing multiple runs you might need to unset the environment before spawning in new processes.  Here's a simple helper function to do that.

```python
def reset_env():
    for key in os.environ.keys():
        if key.startswith("WANDB_"):
            del os.environ[key]
```

**Comment by pierclgr** [comment]
I'm not specifying an id in wandb.init(). I'm trying to use the helper function that you gave to me, I'll keep you updated. Thanks :)

**Comment by MBakirWB** [comment]
Hi @pierclgr,

Reaching back out regarding this support request, do you have any updates or require further assistance?

**Comment by MBakirWB** [comment]
Hi @pierclgr ,

As we have not heard back from you, we are going to close this request. If you would like to re-open the conversation, please let us know!

**Comment by tommasocarraro** [comment]
Hi, I have a similar problem. I have opened this issue https://github.com/wandb/wandb/issues/4653.
From what I have understood, this problem happens if you call wandb.init() after wandb.agent(), for example, to create a new run after the sweep is ended.

For some reason, wandb detects the newly created run as an alias of the last run of the sweep. In fact, they share the same ID. I get the same log info too. It is not possible to change the run id when a sweep is running. But at that point in the code, the sweep should be finished. I think that one way to solve this issue is to have a command after wandb.agent() to say to wandb that the sweep has to be finished at that point of the program. In other words, there should be a way to clean the wandb session and start another run. I did not find how to do that for the moment. The only way for the moment is to have wandb.agent() and wandb.init() in different wandb sessions. I honestly do not know why they do not work in the same session.
